### PR TITLE
Add gcloud-sql subcommand

### PR DESCRIPTION
**Pull request by mw2q**
Provision and deploy Google Cloud SQL for evaluation.

### CODE REVIEW

File: edbdeploy/cli.py [hunk 1]
```diff
@@ -124,6 +132,8 @@ def parse():
             azure_db_parser.print_help()
         elif env.cloud == 'gcloud':
             gcloud_parser.print_help()
+        elif env.cloud == 'gcloud_sql':
```
Comment (jt-edb) [inline edbdeploy/cli.py:135]: `gcloud-sql`

**Review by jt-edb — CHANGES_REQUESTED** [review]
Please see my comments.

File: README.md [hunk 2]
```diff
@@ -126,6 +126,7 @@ $ edb-deployment <CLOUD_VENDOR> <SUB_COMMAND> [<PROJECT_NAME>]
   * `azure`: Microsoft Azure Cloud
   * `azure-db`: Microsoft Azure Database
   * `gcloud`: Google Cloud
+  * `gcloud`: Google Cloud SQL for PostgreSQL
```
[the remark this answers is not in the document]
Comment (mw2q) [reply README.md:129]: Oops, will fix.

File: edbdeploy/cli.py [hunk 3]
```diff
@@ -25,7 +26,7 @@ def error(self, message):
 
 def parse():
     parser = CLIParser(
-        description='EDB deployment script for aws, aws-rds, azure, azure-db and gcloud'
+        description='EDB deployment script for aws, aws-rds, azure, azure-db, gcloud, and gcloud_sql'
```
[the remark this answers is not in the document]
Comment (mw2q) [reply edbdeploy/cli.py:29]: Fixed.

File: edbdeploy/cli.py (hunk 1 above)
Comment (mw2q) [reply edbdeploy/cli.py:135]: Fixed.

File: edbdeploy/commands/gcloud_sql.py [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+    )
+
+    # gcloud-sql configure sub-command options
+    subcommand_parsers['configure'].add_argument(
+        '-a', '--reference-architecture',
+        dest='reference_architecture',
+        choices=ReferenceArchitectureOption.choices,
+        default=ReferenceArchitectureOption.default,
+        metavar='<ref-arch-code>',
+        help=ReferenceArchitectureOption.help
```
[the remark this answers is not in the document]
Comment (mw2q) [reply edbdeploy/commands/gcloud_sql.py:27]: Oops, not at this time.  Will renamed ReferenceArchitectureOptionRDS to ReferenceArchitectureOptionDBaaS and fix that up.

File: edbdeploy/data/terraform/gcloud-sql/environments/security/firewall.tf [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+  network = var.network_name
+
+  allow {
+    protocol = "tcp"
+    ports    = ["7800-7810"]
+  }
+
+  target_tags = [format("%s-%s", var.network_name, "firewall-efm")]
+  source_ranges = [var.source_ranges]
+}
```
[the remark this answers is not in the document]
Comment (mw2q) [reply edbdeploy/data/terraform/gcloud-sql/environments/security/firewall.tf:92]: My bad, I actually thought I removed it...

File: edbdeploy/data/terraform/gcloud-sql/environments/security/firewall.tf [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+  network = var.network_name
+
+  allow {
+    protocol = "tcp"
+    ports    = ["6432"]
+  }
+
+  target_tags = [format("%s-%s", var.network_name, "firewall-pgbouncer")]
+  source_ranges = [var.source_ranges]
+}
```
[the remark this answers is not in the document]
Comment (mw2q) [reply edbdeploy/data/terraform/gcloud-sql/environments/security/firewall.tf:209]: Right, I thought I removed it. :)

File: edbdeploy/project.py [hunk 7]
```diff
@@ -1857,7 +1916,7 @@ def setup_tools(cloud):
                 ),
                 'cloud_vendors': [
                     'aws', 'aws-rds', 'aws-rds-aurora', 'azure', 'gcloud',
-                    'baremetal'
+                    'gbloud-sql', 'baremetal'
```
[the remark this answers is not in the document]
Comment (mw2q) [reply edbdeploy/project.py:1919]: Fixed.

**Review by jt-edb — APPROVED** [review]
LGTM

**Pull request merged**